torch/csrc/distributed/rpc/rref_impl.h

```diff
@@ -218,6 +218,19 @@ class TORCH_API RRef : public RRefInterface {
     return type_;
   }
 
+  // Save the future corresponding to the creation of this RRef on a remote
+  // node. Note that this is only set when processing requests invoked with
+  // rpc.remote. This is only used to get the future corresponding to the rref
+  // for profiling use cases.
+  inline void registerCreatingFuture(std::shared_ptr<FutureMessage> fut) {
+    creatingFuture_ = fut;
```

Comment:
std::move?

Reply:
I'll try that, although I'm not sure if we can take ownership of the future here. In `process_group_agent`, we copy the future into `FutureInfo`, this is for the purpose of keeping it alive by holding a ref until the future is completed. Would moving it nullify the shared_ptr stored by ProcessGroupAgent?

Reply:
> Would moving it nullify the shared_ptr stored by ProcessGroupAgent?

I think no. Because we already copy-by-value when passing in the `shared_ptr` `fut` argument. So this `fut` is different from the one in the call site of this function. Using `std::move` here avoids inc+dec the reference count on the shared ptr. Please correct me if I am wrong.

Reply:
This is right, since we already copy by value when calling `registerCreatingFuture`.